.github/workflows/build-and-publish.yml

```diff
@@ -5,12 +5,16 @@ on:
     secrets:
       PYPI_TOKEN:
           required: true
+    inputs:
+      ubuntu-version:
+        required: true
+        type: string
 
 jobs:
 
   cd-job:
     name: Continuous Delivery
-    runs-on: ubuntu-22.04
+    runs-on: ${{ inputs.ubuntu-version }}
```

Comment:
the ubuntu version is only relevant for tests, and this file is generated by the toolbox and only should be changed for a good reason